### ISSUE TITLE
Tagged imaged to docker hub

### ISSUE THREAD

**Issue by bobrik** [digital-wonderland/docker-elasticsearch-curator#1]
It would be super-cool to have version tags for `digitalwonderland/elasticsearch-curator` in docker hub. Then I can stop worrying about the version I am about to run:

```
docker run --rm -i -t digitalwonderland/elasticsearch-curator:2.0.0 --help
```

Let me know if I should do that myself :)

**Comment by digital-wonderland** [comment]
I agree that this would be totally awesome but unfortunately I don't see how this is currently possible:
1. atm `pip` is used to install whatever is the [latest on pip](https://pypi.python.org/pypi/elasticsearch-curator) _on build time_. Certainly we could install a specific version but then we would have to edit the `Dockerfile` whenever a new version gets released and this doesn't provide tags yet.
2. builds on Dockers hub by default tag a build as `latest`. One can choose to tag builds by _relative path_ or _branch_ but this has to be configured manually - as in one would have to maintain multiple branches.
3. `pip` doesn't support [semantic versioning](http://semver.org) - as in I am not able to say something like _gimme the latest 1.2_ and I would get _1.2.2_ if it is available (please correct me if I am wrong)

Altogether I am fine with creating another branch / build _once we reach a point where Curator X is not compatible anymore with ES <= Y_ but until then it is just easier to press a button and get the latest and greatest (if it is outdated / needs a rebuild please feel free to open an issue or shoot me a mail - see git log).

To sum it up: if you know an elegant solution, please let me know. The basic idea of this container is just to provide the latest curator builds until some incompatibility, then we would have to switch to different versions / tags in separate branches (if there is any need).

FWIW: I switched the container to CentOS 7 so a rebuild got triggered and now it contains Curator 2.0.0

**Comment by bobrik** [comment]
We have some containers that need slightly different build params and solve that with `Dockerfile` template and shell magic. This repo could have the next structure:
- `Dockerfile` — current contents
- `.build` — empty dir with `.gitignore` do hold builds
- `build.sh` — shell script to build every curator version available

Basically, `build.sh` would go to pip, ask which versions of curator are available and trigger build process for each:

```
cat Dockerfile | sed "s/elasticsearch-curator/elasticsearch-curator==$VERSION"
cd .build
docker build -t "digitalwonderland/elasticsearch-curator:$VERSION" .
docker push "digitalwonderland/elasticsearch-curator:$VERSION"
```

When anything in this repo changes docker hub is notified with github hook and builds `latest` tag with the latest version of curator available.

**Comment by digital-wonderland** [comment]
First of all sorry for taking so long to respond.

Regarding the topic: yes, that way it would be possible, but it does not address my two pain points:
1. this would require me running the script to get a tagged build. Currently I just go to docker hub whenever I notice there is a new version and press a button to get the latest & greatest 5 minutes later tagged as `latest`.
2. this would mean every single curator version is tagged (and therefore doesn't get deleted) from docker hub, which would result in a royal waste of bandwidth, build time & disk space on docker hub, which I consider an abuse of their service (hosting every single container someone decides to publish without any limitations - I'm fully aware that this container wouldn't be that big. It is more about 'if everyone would do that').

**To prevent any misunderstanding:** I have absolutely no problem with creating a separate branch / build as soon as curator > X is not compatible anymore with Elasticsearch < Y, but until then I just would like to stick with `latest` to avoid any unnecessary overhead.
